docs/current_docs/api/clients-cli.mdx

```diff
@@ -21,77 +51,119 @@ The `dagger core` command lets you call the core Dagger API, without needing to
+- Call one of the auto-generated Dagger Functions:
 
+    <Tabs>
+    <TabItem value="System shell">
     ```shell
-    dagger call container-echo --string-arg="Welcome to Dagger!" stdout
+    dagger -c 'container-echo "Welcome to Dagger!" | stdout'
     ```
-
-    Here's what you should see:
-
+    </TabItem>
+    <TabItem value="Dagger Shell">
+    ```shell title="Type 'dagger' to start Dagger Shell in interactive mode and enter the command below."
+    container-echo "Welcome to Dagger!" | stdout
+    ```
+    </TabItem>
+    <TabItem value="Dagger CLI">
     ```shell
-    Welcome to Dagger!
+    dagger call container-echo --string-arg="Welcome to Dagger!" stdout
     ```
+    </TabItem>
+    </Tabs>
 
     :::tip
-    When using `dagger call`, all names (functions, arguments, struct fields, etc) are converted into a shell-friendly "kebab-case" style.
+    When using the Dagger CLI, all names (functions, arguments, struct fields, etc) are converted into a shell-friendly "kebab-case" style.
     :::
 
-- Modules don't need to be installed locally. Dagger lets you consume modules from GitHub repositories as well. You can use Dagger Functions from external modules in exactly the same way as you would use them locally.
-
-    Here is an example of calling a Dagger Function from an external module, which scans container images for vulnerabilities.
+- Modules don't need to be installed locally. Dagger lets you consume modules from GitHub repositories and call their Dagger Functions as though you were calling them locally:
 
+    <Tabs>
+    <TabItem value="System shell">
     ```shell
-    dagger -m github.com/jpadams/daggerverse/trivy@v0.5.0 call scan-image --image-ref=ubuntu:latest
+    dagger <<EOF
+    github.com/jpadams/daggerverse/trivy@v0.5.0 |
+      scan-image ubuntu:latest
+    EOF
     ```
-
-    Here's what you should see:
-
+    </TabItem>
+    <TabItem value="Dagger Shell">
+    ```shell title="Type 'dagger' to start Dagger Shell in interactive mode and enter the command below."
+    github.com/jpadams/daggerverse/trivy@v0.5.0 | scan-image ubuntu:latest
+    ```
+    </TabItem>
+    <TabItem value="Dagger CLI">
     ```shell
-    ubuntu:latest (ubuntu 24.04)
-    ============================
-    Total: 10 (UNKNOWN: 0, LOW: 8, MEDIUM: 2, HIGH: 0, CRITICAL: 0)
-    ...
+    dagger -m github.com/jpadams/daggerverse/trivy@v0.5.0 call \
+      scan-image --image-ref=ubuntu:latest
     ```
+    </TabItem>
+    </Tabs>
 
-- The Dagger Functions available in a Dagger module are listed with `dagger functions`.
-
-    Here is an example of listing Dagger Functions in an external module:
+- List all the Dagger Functions available in a module using context-sensitive help:
 
+    <Tabs>
+    <TabItem value="System shell">
     ```shell
-    dagger -m github.com/jpadams/daggerverse/trivy@v0.5.0 functions
+    dagger -c 'github.com/jpadams/daggerverse/trivy@v0.5.0 | .help'
```

Comment:
For documenting a module `.help <module>` should be used. However it also includes constructor/entrypoint usage, so not quite the same as `dagger functions`.

Also, if the module does have required arguments just `<module> | .help` will return an error because you need to actually call the constructor in order to get its return with `<module> | .help` like this.

While you generally should do `<function> | .help` to get available functions after calling another function, for a module I suggest preferring to start with `.help <module>` instead.

Reply:
I tried `.help github.com/jpadams/daggerverse/trivy@v0.5.0` as you suggested but it doesn't produce the list of available functions (which is the task I'm trying to solve for in this bullet point).

Reply:
It's in a fix in main. Will be released in v0.17. If you want to try it you need a dev build.